Parser function's unit test

CaioCordeiro opened the pull request:
This Pull Request adds unit tests to all parser.h functions.
Changes:
- Adds unit test;
- Adds the unit test compilation script together with the
expected output to the "Makefile.am" so it can be runned with the
'make check' command.
- Adds the test documentation to the README.

CaioCordeiro committed:
Add parser unit test

This patch adds unit tests to all parser.h functions.
Functions tested:
- parser_config_create_object;
- parser_update_to_list;
- parser_data_create_object;
- parser_config_to_list;
- parser_queue_from_json_array;
- parser_request_to_list;
- parser_sensorid_to_json;
- parser_device_json_create;
- parser_auth_json_create;
- parser_unregister_json_create;
- parser_get_key_str_from_json_str;
- parser_is_key_str_or_null;

CaioCordeiro committed:
Add unit test to automake

This patch adds the unit test compilation script together with the
expected output to the "Makefile.am" so it can be runned with the
'make check' command.

CaioCordeiro committed:
Update README with the new test command

This patch adds the test documentation to the README.

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+static int test_schema_create_object(void)
+{
+	struct l_queue *queue;
+
```
Is this blank line necessary?

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	struct l_queue *queue;
+
+	const char *json_str;
+
```
Is this blank line necessary?

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	" \"valueType\": 3, \"unit\": 0, \"typeId\": 65521, "
+	"\"name\": \"Schema_test\" } ] }", json_str);
+
+	if(err != 0 ) {
```
```suggestion
	if(err != 0) {
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	const char *json_str;
+	int err;
+
+	json_str = parser_data_create_object("DEVICE_1", 12,1, &value,
```
```suggestion
	json_str = parser_data_create_object("DEVICE_1", 12, 1, &value,
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	err = strcmp("{ \"id\": \"DEVICE_1\", \"data\": [ { \"sensorId\": 12,"
+		     " \"value\": 1223 } ] }", json_str);
+
+	if(err != 0 ) {
```
```suggestion
	if(err != 0) {
```

luigiluz replied:
This happens a few times through the entire code, please change it on the other occurrences too

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+					"\"test_device_id\", \"token\":"
+					" \"test_device_token\" }", "TEST_KEY");
+
+	if(!res) {
+
+		return -EINVAL;
+	}
```
```suggestion
	if(!res)
		return -EINVAL;
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+
+	msg = l_queue_pop_head(queue);
+
+	if(msg->sensor_id != 1 || msg->payload.val_i != 0) {
+		goto error;
+	}
```
```suggestion
	if(msg->sensor_id != 1 || msg->payload.val_i != 0)
		goto error;
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+
+	msg = l_queue_pop_head(queue);
+
+	if(msg->sensor_id != 2 || msg->payload.val_i != 1000) {
+		goto error;
+	}
```
```suggestion
	if(msg->sensor_id != 2 || msg->payload.val_i != 1000)
		goto error;
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	l_queue_destroy(queue, l_free);
+	l_free(msg);
+
+	return 0;
+error:
```
```suggestion
	return 0;

error:
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	l_queue_destroy(queue, l_free);
+	l_free(sensor_id);
+
+	return 0;
+error:
```
```suggestion
	return 0;

error:
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	res = 0;
+	res_p = &res;
+
+	if(strcmp(name, "test") != 0 )
```
```suggestion
	if(strcmp(name, "test") != 0)
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	if(strcmp(name, "test") != 0 )
+		goto error;
+
+	if(strcmp(id, "TEST_ID_1") != 0 )
```
```suggestion
	if(strcmp(id, "TEST_ID_1") != 0)
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	return -EINVAL;
+}
+
+static void *check_json_integrity(const char *id, const char *name,
+		struct l_queue *schema)
```
Please be sure that the struct l_queue parameter is alligned with const char *id

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	res = -EINVAL;
+
+	return l_memdup(res_p, sizeof(int));
+
```
Please remove this blank line

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+
+	err = l_queue_pop_head(queue);
+
+	if(err == NULL) {
+		goto error;
+	}
```
```suggestion
	if(err == NULL)
		goto error;
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+		goto error;
+	}
+
+	if(*err < 0) {
+		goto error;
+	}
```
```suggestion
	if(*err < 0)
		goto error;
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	l_queue_destroy(queue, l_free);
+	l_free(err);
+
+	return 0;
+error:
```
```suggestion
	return 0;
	
error:
```

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	if (err < 0)
+		l_error("REQUEST_TO_LIST: ERR");
+	else
+		l_info("REQUEST_TO_LIST: OK");
+	err = test_queue_from_json_array();
```
```suggestion
		l_info("REQUEST_TO_LIST: OK");
		
	err = test_queue_from_json_array();
```

CaioCordeiro replied:
Done :rocket:

CaioCordeiro replied:
Done :rocket:

luigiluz commented on test/unit/parser/parser_unit_test.c:
```diff
@@ -0,0 +1,440 @@
+	if (err < 0)
+		l_error("UNREGISTER_JSON_CREATE: ERR");
+	else
+		l_info("UNREGISTER_JSON_CREATE: OK");
+	err = test_get_key_str_from_json_str();
```
```suggestion
		l_info("UNREGISTER_JSON_CREATE: OK");
		
	err = test_get_key_str_from_json_str();
```

CaioCordeiro replied:
Done :rocket:

CaioCordeiro replied:
Done :rocket:

luigiluz reviewed (changes requested):
Nice PR! Just made a few remarks on coding style

CaioCordeiro committed:
Fix create_device_item_cb parameter name

This patch changes the "struct l_queue *schema" parameter to
"struct l_queue *config" to match the code terminology.
